### PR TITLE
Feat/test: Admin merchant enable disable

### PR DESCRIPTION
**Pull request by PhiMed**

### CODE REVIEW

**Review by GregoryJFischer — COMMENTED** [review]
DS store should not be included in the github repo I think, maybe add it to got ignore?

File: spec/features/admin/merchants/index_spec.rb [hunk 1]
```diff
@@ -27,4 +27,32 @@
 
     expect(current_path).to eq("/admin/merchants/#{@merchant_1.id}")
   end
-end
\ No newline at end of file
+
+  it 'has an interface to enable or disable merchants' do
+    @merchant_2.destroy
```
Comment (GregoryJFischer) [inline spec/features/admin/merchants/index_spec.rb]: I think that the destroy needs to be there means that there is something else going on... can't tell at the moment, but we should go through it and try to figure out what's going on

File: app/views/admin/merchants/index.html.erb [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
     </li>
   <% end %>
-</ul>
\ No newline at end of file
+</ul>
+
+<h3>Disabled Merchants</h3>
+<ul id="disabled_merchants">
+  <% @merchants.disabled.each_with_index do |merchant, index| %>
+    <li id="merchant_<%=index%>">
```
Comment (yosoynatebrown) [inline app/views/admin/merchants/index.html.erb]: I think if we change it to id="merchant_<%= merchant.id %>" we could have CSS ids that match up with specific merchants for testing

File: spec/features/admin/merchants/index_spec.rb [hunk 3]
```diff
@@ -27,4 +27,32 @@
...
+    @merchant_3.destroy
+    @merchant_4.destroy
+    @merchant_5.destroy
+
+    visit admin_merchants_path
+
+    within('#disabled_merchants') do
+      expect(page).not_to have_button("Disable")
```
Comment (yosoynatebrown) [inline spec/features/admin/merchants/index_spec.rb]: we could use a `within "merchant_#{merchant.id}"` block to test if a specific merchant is being changed

File: app/controllers/admin/merchants_controller.rb [hunk 4]
```diff
@@ -6,4 +6,26 @@ def index
...
+     if params[:enable]
+       merchant.update!(status: 1)
+       redirect_to "/admin/merchants"
+     elsif params[:disable]
+       merchant.update!(status: 0)
+       redirect_to "/admin/merchants"
+
+     else
+       if merchant.update!(merchant_params)
+         flash[:alert] = "Information has been successfully updated"
```
Comment (yosoynatebrown) [inline app/controllers/admin/merchants_controller.rb:22]: Like that you included flash messages.

File: spec/features/admin/merchants/index_spec.rb [hunk 5]
```diff
@@ -27,4 +27,32 @@
...
+
+      click_button("Enable")
+
+      expect(page).not_to have_button("Disable")
+      expect(page).not_to have_content(@merchant_1.name)
+    end
+
+    within('#enabled_merchants') do
```
Comment (GregoryJFischer) [inline spec/features/admin/merchants/index_spec.rb]: I think this is the wrong block for the within. We want to see the specific enabled merchant not all of them, maybe this is why you have to disable the others?

**Review by GregoryJFischer — COMMENTED** [review]
Need to refactor within blocks probably

**Comment by GregoryJFischer** [comment]
Awesome job, looks good!

**Pull request merged**